devicetypes/smartthings/zigbee-metering-plug.src/zigbee-metering-plug.groovy

```diff
@@ -0,0 +1,136 @@
+                state "default", label:'reset kWh', action:"reset"
+        }
+
+        main(["switch"])
+        details(["switch","power","energy","refresh","reset"])
+    }
+}
+
+def getATTRIBUTE_READING_INFO_SET() { 0x0000 }
+def getATTRIBUTE_HISTORICAL_CONSUMPTION() { 0x0400 }
+
+def parse(String description) {
+    log.debug "description is $description"
+    def event = zigbee.getEvent(description)
+    if (event) {
+        log.info "event enter:$event"
+        if (event.name== "power") {
+            event.value = event.value
+            event.unit = "W"
+        } else if (event.name== "energy") {
+            event.value = event.value/100
+            event.unit = "kWh"
+        }
+        log.info "event outer:$event"
+        sendEvent(event)
+    } else {
+        List result = []
+        def descMap = zigbee.parseDescriptionAsMap(description)
+        log.debug "Desc Map: $descMap"
+        
+        List attrData = [[clusterInt: descMap.clusterInt ,attrInt: descMap.attrInt, value: descMap.value]]
+        descMap.additionalAttrs.each {
+            attrData << [clusterInt: descMap.clusterInt, attrInt: it.attrInt, value: it.value]
+        }
+        attrData.each {
+                def map = [:]
+                if (it.clusterInt == zigbee.SIMPLE_METERING_CLUSTER && it.attrInt == ATTRIBUTE_HISTORICAL_CONSUMPTION) {
+                        log.debug "power"
+                        map.name = "power"
+                        map.value = zigbee.convertHexToInt(it.value)/1000
+                        map.unit = "W"
+                }
+                else if (it.clusterInt == zigbee.SIMPLE_METERING_CLUSTER && it.attrInt == ATTRIBUTE_READING_INFO_SET) {
+                        log.debug "energy"
+                        map.name = "energy"
+                        map.value = zigbee.convertHexToInt(it.value)/100
+                        map.unit = "kWh"
+                }
+
+                if (map) {
+                        result << createEvent(map)
+                }
+                log.debug "Parse returned $map"
+        }
+        return result       
+    }
+}
+
+def off() {
+    zigbee.off()
+}
+
+def on() {
+    zigbee.on()
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+    return refresh()
+}
+
+def refresh() {
+    log.debug "refresh"
+    zigbee.electricMeasurementPowerRefresh() +
+           zigbee.readAttribute(zigbee.SIMPLE_METERING_CLUSTER, 0x0000) +
+           zigbee.onOffRefresh()
+}
+
+def configure() {
+    sendEvent(name: "checkInterval", value: 10 * 60 + 1 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+    log.debug "Configuring Reporting"
+
+    return refresh() +
+           zigbee.onOffConfig() +
+           zigbee.configureReporting(zigbee.SIMPLE_METERING_CLUSTER, 0x0000, DataType.UINT48, 1, 600, 1) +
```

Comment:
It looks like this device supports both Simple Metering and Electrical Measurement and you're configuring both even though they end up mapping to the same attributes. Can you help me understand why you chose to go that route?